File: android/src/main/kotlin/com/example/design_engine/layer2_core/light/EOnSecondaryFixedVariantLight.kt
package com.example.design_engine.layer2_core.light
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object EOnSecondaryFixedVariantLight {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyl32
            AppTheme.STANDARD -> R.color.cstl32
            AppTheme.BLUE -> R.color.cbl32
            AppTheme.GREEN -> R.color.cgl32
            AppTheme.RED -> R.color.crl32
            AppTheme.NOTHING -> R.color.cnol32
        }
    }
}

File: lib/layer2_core/light/e_on_secondary_fixed_variant_light.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/light/system.dart';
import '../../layer1_raw/light/standard.dart';
import '../../layer1_raw/light/blue.dart';
import '../../layer1_raw/light/green.dart';
import '../../layer1_raw/light/red.dart';
import '../../layer1_raw/light/nothing.dart';

class EOnSecondaryFixedVariantLight {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyl32;
      case 'standard': return cstl32;
      case 'blue': return cbl32;
      case 'green': return cgl32;
      case 'red': return crl32;
      case 'nothing': return cnol32;
      default: return cstl32;
    }
  }
}
